### PR TITLE
List, create, delete modules

### PR DESCRIPTION
**Pull request by formulahendry**

### CODE REVIEW

File: src/iotHubModuleExplorer.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+            if (err) {
+                this.outputLine(label, `Error: ${err.message}`);
+                TelemetryClient.sendEvent(Constants.IoTHubAICreateModuleDoneEvent, { Result: "Fail", Message: err.message });
+            } else {
+                this.outputLine(label, `Created: ${JSON.stringify(module, null, 2)}`);
+                vscode.commands.executeCommand("azure-iot-toolkit.refresh", deviceItem);
+                TelemetryClient.sendEvent(Constants.IoTHubAICreateModuleDoneEvent, { Result: "Success" });
+            }
+        });
+    }
```
Comment (LazarusX) [inline src/iotHubModuleExplorer.ts]: We might need to refresh the device tree after creating modules.

File: src/iotHubModuleExplorer.ts [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        this._outputChannel.show();
+        this.outputLine(label, `Creating '${moduleId}'`);
+
+        registry.addModule({ deviceId: deviceItem.deviceId, moduleId }, (err, module) => {
+            if (err) {
+                this.outputLine(label, `Error: ${err.message}`);
+                TelemetryClient.sendEvent(Constants.IoTHubAICreateModuleDoneEvent, { Result: "Fail", Message: err.message });
+            } else {
+                this.outputLine(label, `Created: ${JSON.stringify(module, null, 2)}`);
+                vscode.commands.executeCommand("azure-iot-toolkit.refresh", deviceItem);
```
Comment (LazarusX) [inline src/iotHubModuleExplorer.ts]: My module list is not refreshed after creating new modules. Can you reproduce on your side?

My steps:
- Expand the module list of a device
- Create a new module
- The module list is not refreshed.

File: src/iotHubModuleExplorer.ts [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    }
+
+    public async createModule(deviceItem?: DeviceItem) {
+        const label = "Module";
+        deviceItem = await Utility.getInputDevice(deviceItem, Constants.IoTHubAICreateModuleStartEvent);
+        if (!deviceItem) {
+            return;
+        }
+
+        const iotHubConnectionString = await Utility.getConnectionString(Constants.IotHubConnectionStringKey, Constants.IotHubConnectionStringTitle);
```
Comment (adashen) [inline src/iotHubModuleExplorer.ts]: Do we need hub string to create module? I think device connection string should be enough

File: src/iotHubModuleExplorer.ts [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+                vscode.commands.executeCommand("azure-iot-toolkit.refresh", deviceItem);
+                TelemetryClient.sendEvent(Constants.IoTHubAICreateModuleDoneEvent, { Result: "Success" });
+            }
+        });
+    }
+
+    public async deleteModule(moduleItem: ModuleItem) {
+        TelemetryClient.sendEvent(Constants.IoTHubAIDeleteModuleStartEvent);
+        const label = "Module";
+        const iotHubConnectionString = await Utility.getConnectionString(Constants.IotHubConnectionStringKey, Constants.IotHubConnectionStringTitle);
```
Comment (adashen) [inline src/iotHubModuleExplorer.ts]: same as above

File: src/iotHubModuleExplorer.ts (hunk 3 above)
Comment (formulahendry) [reply src/iotHubModuleExplorer.ts]: It is needed, It is in service SDK: `const registry: iothub.Registry = iothub.Registry.fromConnectionString(iotHubConnectionString);`

**Pull request merged**